src/main/java/com/uid2/admin/auth/AdminKeyset.java

```diff
@@ -0,0 +1,77 @@
+package com.uid2.admin.auth;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.uid2.admin.model.ClientType;
+import com.uid2.shared.auth.Keyset;
+import org.checkerframework.checker.units.qual.K;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class AdminKeyset {
+
+    @JsonProperty("allowed_types")
+    private final Set<ClientType> allowedTypes;
+
+    private final Keyset keyset;
+
+    public AdminKeyset(int keysetId, int siteId, String name, Set<Integer> allowedSites, long created,
```

Comment:
I see dozens of `new HashSet<>` throughout this PR
Consider having a constructor that doesn't need allowedTypes so it can do the `new HashSet<>` internally

Reply:
Fixed and replaced all the instants back to what they were